cmd/catalyst-node/handlers/access-control/access-control.go

```diff
@@ -119,7 +118,13 @@ func (ac *PlaybackAccessControl) handleUserNew(payload []byte) []byte {
 		pubKey = claims.PublicKey
```

Comment:
It would be even better to have this `jwt` logic (`pubKey` extraction) into a re-usable helper as well. Perhaps the helper gets the playback URL and it tells whether the playback is authorized or not. This would allow both assets and streams to have a consistent access control experience, being able to use jwt or webhooks on either.

This is not a blocker if we're running tight on schedule, but I think it would avoid creating a tech debt item which would be having livestream and VOD access control experiences completely different (instead of both types supporting both protocols).

WDYT? Does it involve a too big refactor to be more worth doing only as a follow up maybe next month?

Reply:
i think that makes sense to me, so you're thinking this kind of thing: https://github.com/livepeer/catalyst/pull/437/commits/527686c30e4009de9ac07c5fb0a6a3ce2815b361

Reply:
Yeah exatly! I just noticed 1 thing on that commit: you're using the `tkn` query-string for the access key, which is something that already exists on the mist side and AFAIK has a different meaning. We should avoid that potential conflict by using something else. @iameli should have more context of this.

I had suggested an explicit `accessKey` query-string instead, which also gains in readability IMO.

Reply:
@victorges could you give this another check if you're happy with the final version of this function. I've tested integrated it locally into catalyst-api and working 👍

Reply:
@mjh1 Checked the new version and looks good! Made some minor comments but overall already LGTM!